### PR TITLE
Refactor cowell

### PR DESCRIPTION
**Pull request by astrojuanlu**
See discussion in gh-1052. Did not run the "slow" tests or the notebooks, they will probably fail.

### CODE REVIEW

**Comment by astrojuanlu** [comment]
As expected, MyPy and the slow tests (including the docs) failed. But I actually like how this simplifies the code, and the burden on the user side is only moderate. We could add a helper to assemble several perturbations together (or just leave it like that).

**Comment by jorgepiloto** [comment]
Congrats on this, the implementation is very clean. I really like this new approach, as now the `func_twobody` is the "true" two-body function. In addition, looks like all the test passed :clap:

File: src/poliastro/twobody/events.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def terminal(self):
     def last_t(self):
         return self._last_t * u.s
 
-    def __call__(self, t, u):
+    def __call__(self, t, u, k):
```
Comment (jorgepiloto) [inline src/poliastro/twobody/events.py:28]: Is this `k` ever used? :thinking:

**Review by jorgepiloto — CHANGES_REQUESTED** [review]
Just left one comment on a parameter which seems not to be used inside a `__call__` method. All the rest modifications look good by my side :+1:

File: src/poliastro/twobody/events.py (hunk 1 above)
Comment (astrojuanlu) [reply src/poliastro/twobody/events.py:28]: Not in the `LithobrakeEvent`, but potentially in other events! Also, makes the calling convention of the objective function and the events consistent.

**Comment by astrojuanlu** [comment]
Merging! Welcome to the future 🚀

**Pull request merged**